### ISSUE TITLE
AS0011 Compiler Error for Enum Extension

### ISSUE THREAD

**Issue by ralf-escher** [microsoft/AL#6824]
Starting this afternoon we get an AS0011 compiler error for an existing Enum Extension.
`AL Compiler version 8.1.8.16322`

Our Enum values don't have a mandatory affix, therefore our app does not longer compile successfully (it did before).

```
enumextension 5042650 "BET IND Comment Line Table Name" extends "Comment Line Table Name"
{
    value(5042650; "Packaging Structure Definition Header") { }
    value(5042651; "Packaging Structure Definition Line") { }
}
```

I tried to mark those Enum values as Obsolete, but the error still occurs.
I also tried to remove the Enum Extension or to remove the invalid Enum Values, but then our build pipeline (based on the CI/CD example from @freddydk) using `Run-AlPipeline` shows a breaking change, since we have removed it.

_error AS0083: The enum value with name 'Packaging Structure Definition Header' and ID '5042650' was found in the previous version of 'EnumExtension "BET IND Command Line Table Name"', but is missing in the current extension. This might break the upgrade of existing installations and dependent extensions._

So... I can not fix it.

My current solution is using pragma:
_(I also refer to a different Enum, therefore I could build a new correct Enum Extension)_
```
enumextension 5042650 "BET IND Comment Line Table Name" extends "Service Comment Line Type"
{
#pragma warning disable AS0011
    value(5042650; "Packaging Structure Definition Header")
    {
        ObsoleteState = Pending;
        ObsoleteReason = 'Prefix missing';
        ObsoleteTag = '31.0';
    }
    value(5042651; "Packaging Structure Definition Line")
    {
        ObsoleteState = Pending;
        ObsoleteReason = 'Prefix missing';
        ObsoleteTag = '31.0';
    }
#pragma warning restore
}
```

**Comment by dnpb** [comment]
I thinks this Yammer thread covers it:
https://www.yammer.com/dynamicsnavdev/#/Threads/show?threadId=1415324861079552

The upshot [was](https://www.yammer.com/dynamicsnavdev/#/Threads/show?threadId=1415324861079552):
> We discussed this internally and we won't enforce the rule on existing violations due to the manual cost and the risk that updating enum value IDs and names can be imply.
>
>For newly introduced enum values without affixes (they don't exist in the baseline version), we will report an error with AS0011. This way we prevent more violations to be added.
>
>The rule AS0098 will then be reporting a warning on enum values which do not have affixes and that were already defined without affixes in the baseline version. It is kept as a warning for awareness that the enum names could collide one day with the enum values defined in another app.

**Comment by dnpb** [comment]
Btw, you'll also run into warnings and potential runtime issues with enumerator names that long; see
https://github.com/microsoft/AL/issues/6740#issuecomment-911443709

**Comment by qutreson** [comment]
This issue should be now fixed in version 8.2.545335 of the AL Language extension.

The behavior of the affix validation is now as follows for enum values without affixes defined in enum extensions:
- if you don't have a baseline set in your AppSourceCop.json, AS0098 is reported (warning)
- if you have a baseline and your enum value is in the baseline without affix, AS0098 is reported (warning)
- if you have a baseline and your enum value is not in the baseline, AS0011 is reported (error)
